### ISSUE TITLE
aspect-ratio media queries

### ISSUE THREAD

**Issue by amarchen** [alpacaaa/react-native-media-queries#2]
You've got an awesome utility! Simplifies the layout tuning a lot.

One thing I need quite a lot in my app is portrait-landscape adjustments. E.g. some elements should be hidden completely in landscape (status bar) or resized very heavily. Unlike on the web exactly portrait-landscape adjustment IMHO is quite frequent in the mobile apps.

I am using min-height queries to achieve the variation at the moment. It would have been simpler if aspect-ratio queries were supported.

**Comment by alpacaaa** [comment]
Thanks @amarchen! Your comment is spot on, to be honest I didn't know `aspect-ratio` was present in the spec, never used it before. But sounds extremely useful, I'm using `minHeight` as well but it doesn't always feels right, so I think I'm going to add support for something along the lines of `aspect-ratio`.

**Comment by amarchen** [comment]
Awesome! Good luck with it!

There are lots of useful media query ways including even number of physical pixels per CSS pixel that are outlined well at https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries

When it comes to react-native apps, I guess one that could also be particularly useful could be `orientation` - https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#orientation

Note that it's not the same as `device-aspect-ratio >1`, it operates on the visible viewport (same as just `aspect-ratio`?), so if a portrait device opens a software keyboard, it can change the visible viewport to landscape from portrait.

Not sure if you are interested in replicating the media queries that much. Possibly `device-aspect-ratio` would be good enough.

P.S.
I just realized that at the moment `react-native-media-queries` is using `height` while probably meaning `device-height` as I don't think it adjusts for software keyboard status.

**Comment by alpacaaa** [comment]
Media queries with `aspect-ratio` functionality are now available!

I haven't written any documentation yet but it should be fairly straightforward. The functions exposed are:
- `maxAspectRatio`
- `minAspectRatio`
- `aspectRatio`

You can just pass in any ratio and it'll do the math, e.g.: `aspectRatio(16 / 9, { ...your styles })`. Let me know if it works!

I don't think `device-aspect-ratio` is necessary as it will always be the same on mobile and it probably wouldn't matter on desktop. 

Also, the keyboard suggestion is interesting, but I want to make sure to get it right. I don't think going from `portrait` to `landscape` when the keyboard is open would be enough to cover all cases. Maybe we should introduce an extra media query like `keyboardIs('open')` so that you could compose it with other media queries and have more fine grained control on what's happening. You'll end up with something like this:

```

createStyles(
  keyboardIs('open', maxHeight(500, { .. })),

  // or even just
  keyboardIs('open', { ...styles })
)
```

What do you think?

**Comment by alpacaaa** [comment]
This is now released in `0.3.0`.
